### ISSUE TITLE
Water mesh is too small

### ISSUE THREAD

**Issue by IstuntmanI** [Pottus/ColAndreas#29]
In "src/WaterArray.h":
```C++
{ -10000.0, 10000.0, 0.0, 3000.0, 10000.0, 0.0, -10000.0, 3000.0, 0.0 },
{ 3000.0, 10000.0, 0.0, 10000.0, 10000.0, 0.0, 10000.0, -3000.0, 0.0 },
{ 10000.0, -3000.0, 0.0, 10000.0, -10000.0, 0.0, -3000.0, -10000.0, 0.0 },
{ -10000.0, 3000.0, 0.0, -10000.0, -10000.0, 0.0, -3000.0, -10000.0, 0.0 },
{ -10000.0, 3000.0, 0.0, 3000.0, 10000.0, 0.0, 3000.0, 3000.0, 0.0 },
{ 3000.0, 10000.0, 0.0, 10000.0, -3000.0, 0.0, 3000.0, -3000.0, 0.0 },
{ -3000.0, -3000.0, 0.0, 10000.0, -3000.0, 0.0, -3000.0, -10000.0, 0.0 },
{ -10000.0, 3000.0, 0.0, -3000.0, 3000.0, 0.0, -3000.0, -10000.0, 0.0 }
```
All those +- 10000 values should be set to infinity. By default, the world boundaries are ( -20000, -20000, 20000, 20000 ), so players could get behind those +- 10000 values. World boundaries could be increased, so I think that this would be ok:
```C++
{ - ( ( float ) 0x7F800000 ), ( float ) 0x7F800000, 0.0, 3000.0, ( float ) 0x7F800000, 0.0, - ( ( float ) 0x7F800000 ), 3000.0, 0.0 },
{ 3000.0, ( float ) 0x7F800000, 0.0, ( float ) 0x7F800000, ( float ) 0x7F800000, 0.0, ( float ) 0x7F800000, -3000.0, 0.0 },
{ ( float ) 0x7F800000, -3000.0, 0.0, ( float ) 0x7F800000, - ( ( float ) 0x7F800000 ), 0.0, -3000.0, - ( ( float ) 0x7F800000 ), 0.0 },
{ - ( ( float ) 0x7F800000 ), 3000.0, 0.0, - ( ( float ) 0x7F800000 ), - ( ( float ) 0x7F800000 ), 0.0, -3000.0, - ( ( float ) 0x7F800000 ), 0.0 },
{ - ( ( float ) 0x7F800000 ), 3000.0, 0.0, 3000.0, ( float ) 0x7F800000, 0.0, 3000.0, 3000.0, 0.0 },
{ 3000.0, ( float ) 0x7F800000, 0.0, ( float ) 0x7F800000, -3000.0, 0.0, 3000.0, -3000.0, 0.0 },
{ -3000.0, -3000.0, 0.0, ( float ) 0x7F800000, -3000.0, 0.0, -3000.0, - ( ( float ) 0x7F800000 ), 0.0 },
{ - ( ( float ) 0x7F800000 ), 3000.0, 0.0, -3000.0, 3000.0, 0.0, -3000.0, - ( ( float ) 0x7F800000 ), 0.0 }
```

By the way, I downloaded the latest colandreas.inc from the Releases page and it is bugged. Also, files should come in an archive, just like the previous version.

**Comment by Crayder** [comment]
>By the way, I downloaded the latest colandreas.inc from the Releases page and it is bugged. Also, files should come in an archive, just like the previous version.

Yeah for some reason GitHub keeps appending the old include when I upload a new one. Even when I delete and reupload it. I've tried many times!

**Comment by Crayder** [comment]
And about the water, not sure if infinite polygons are a great idea. If anything I think we should set them to `20000.0`. That is the boundary we should all stay in no matter what.

**Comment by IstuntmanI** [comment]
Well, you could test infinite polygons for the water mesh. This would be the proper way to do it, as the margin water is most probably infinite in GTA. I think that bulletPhysics supports that. If you won't make it infinite, make it ```30000.0```, just in case. I don't think too many developers extended the world boundaries past ```20000.0```. Those boundaries aren't a solid wall, but something that is pushing you back, so the limit should be at least ```20050.0``` I guess (maybe you could push the limit a bit with an airplane ? didn't test that).

**Comment by Crayder** [comment]
Players can go past the boundary no problem, the problem is the syncing bugs and slight side effects that occur past 20000. It's best to stick to these boundaries not to encourage going past it. Bullet can most definitely support infinite it's just not really recommended within SA-MP. I'd be fine with setting the boundary to 20000 though.
